spark/src/main/java/org/apache/iceberg/RemoveOrphanFilesActionResult.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.List;
+
+public class RemoveOrphanFilesActionResult {
```

Comment:
Why return a result class when you could just return `List<String>`? Do we need these wrappers?

Reply:
That's something we need to discuss. The idea behind this was to report deleted metadata and data files separately when we extend this API in the future. If we want to report all deleted files together or if we want to provide a separate action for metadata, we can return `List<String>`. 

Since it is a new API, we need to decide what actions should report back. For example, do actions need to return a summary? In `RewriteManifestsAction`, this can be a list of manifests that were replaced and a list of manifests that were added. People can infer that themselves but it will require more steps. They would need to assign a label to the produced snapshot, load that snapshot and then access the snapshot summary which will contain only numbers.

Reply:
If this will return two lists in the future, then I think it makes sense to keep the class. Thanks!

Reply:
I removed the wrapper as it would require additional complexity to report data and metadata files separately. Users can call the action on data and metadata separately if they need such functionality.